angular-template.js

```diff
@@ -7,5 +7,28 @@ module.exports = {
             Overrides to Angular template recommended rules:
             https://github.com/angular-eslint/angular-eslint/blob/master/packages/eslint-plugin-template/src/configs/recommended.json
         */
+
+        /*
+            Overrides to Angular rules outside of the recommended configuration:
+        */
+
+        '@angular-eslint/template/accessibility-alt-text': 'error',
+        '@angular-eslint/template/accessibility-elements-content': 'error',
+        '@angular-eslint/template/accessibility-label-for': 'error',
+        '@angular-eslint/template/accessibility-label-has-associated-control': 'error',
+        '@angular-eslint/template/accessibility-table-scope': 'error',
+        '@angular-eslint/template/accessibility-valid-aria': 'error',
+        '@angular-eslint/template/click-events-have-key-events': 'error',
+        '@angular-eslint/template/conditional-complexity': 'error',
+        '@angular-eslint/template/cyclomatic-complexity': 'error',
+        '@angular-eslint/template/i18n': 'error',
+        '@angular-eslint/template/mouse-events-have-key-events': 'error',
+        '@angular-eslint/template/no-any': 'error',
+        '@angular-eslint/template/no-autofocus': 'error',
+        '@angular-eslint/template/no-call-expression': 'error',
+        '@angular-eslint/template/no-distracting-elements': 'error',
+        '@angular-eslint/template/no-duplicate-attributes': 'error',
+        '@angular-eslint/template/no-positive-tabindex': 'error',
+        '@angular-eslint/template/use-track-by-function': 'error'
```

Comment:
This likely applies to certain applications.

**Proposal**
Leave this rule disabled.

Reply:
Wow, the docs are terrible. "Ensures trackBy function is used. Rationale: The use of ‘trackBy’ is considered a good practice." 🙄

This [blog](https://netbasal.com/angular-2-improve-performance-with-trackby-cc147b5104e5) gives a decent summary. If you're using an `ngFor` to create elements, when the collection you're iterating changes Angular can't tell which items were added/removed so it has to recreate all the elements. A `trackBy` function lets you provide a unique identifier for each item so Angular can reuse elements.

I think this could be a useful optimization but agree we shouldn't require it for every `ngFor`.

Reply:
Every NgFor tracks by reference by default. Therefore, it can tell what was added and removed if the references weren't changed. This isn't so helpful if your array is populated by a request that creates new references.

Reply:
@TrevorKarjanis this is ready to close